src/corona_hakab_model/sparse_matrix/sparse.hpp

```diff
@@ -0,0 +1,68 @@
+#include <map>
+#include <vector>
+#include <tuple>
+#include <unordered_set>
+
+typedef float dtype;
+
+// todo benchmark
+
+class MagicOperator{
+    public:
+        //invariant: if w_val = 0 or v_val = 0, operate(w_val, v_val) = 1
+        virtual dtype operate(dtype w_val, dtype v_val) const;
+};
+
+class ManifestMatrix;
+
+class SparseMatrix{
+    private:
+        // note, each column set only stores indices that might not be zero, the only guarantee
```

Comment:
this is relevant to the entice cpp codebase, but why not use ptr objects? unique_ptr and whared ptr?
this feels like an opportunity for memory leakage

Reply:
Swig's support for smart pointers is limited, There aren't very many allocations throughout the code (aside from ctors and dtors)